x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/InferenceCrudIT.java

```diff
@@ -182,6 +183,25 @@ public void testDeleteEndpointWhileReferencedBySemanticText() throws IOException
             assertThat(entityString, containsString("\"acknowledged\":true"));
         }
         deleteIndex(indexName);
+
+        putModel(searchEndpointId, mockSparseServiceModelConfig(), TaskType.SPARSE_EMBEDDING);
+        putSemanticText(endpointId, searchEndpointId, indexName);
+        {
+            var e = expectThrows(ResponseException.class, () -> deleteModel(searchEndpointId));
+            assertThat(e.getMessage(), containsString(buildErrorString.apply(searchEndpointId)));
+        }
+        {
+            var response = deleteModel(searchEndpointId, "dry_run=true");
+            var entityString = EntityUtils.toString(response.getEntity());
+            assertThat(entityString, containsString("\"acknowledged\":false"));
+            assertThat(entityString, containsString(indexName));
```

Comment:
should we check for the endpoint IDs here as well?

Reply:
sure, we can check that